plugins/opds.koplugin/opdsbrowser.lua

```diff
@@ -794,7 +842,11 @@ function OPDSBrowser:showDownloads(item)
             end
             if filetype then -- supported file type
                 local text = acquisition.title and acquisition.title or string.upper(filetype)
-                table.insert(type_buttons, {
+
+                -- Got the url decode from here, https://bit.ly/3F0h6sO
```

Comment:
No need for URL shorteners
```suggestion
                -- Got the url decode from here, https://help.interfaceware.com/code/details/urlcode-lua
```

But we actually have the same principle already:
https://github.com/koreader/koreader/blob/ee35257f997ec3cc719d23212428d39b4173129d/reader.lua#L95-L109

Reply:
I Tried this function and it didn't work, is there any particular reason we should use this function over what's implemented currently?

Also, I will change the URL to it's unshortened version.

Reply:
That was mostly just an aside, though generically speaking the answer would be maintainability. Doing the same thing in different ways makes it harder to work with the codebase. But what exactly do you mean by didn't work? If you mean the obvious difference with the + then there's a chance that's a bug. If you mean the file:// stuff that's not really relevant here. :-)

Reply:
Ah, so that's my bad :) I just copied the whole function into opdsbrowser to see if it works, but I must have either used the function wrong or misunderstood the components. Let me try again, and see if I can make it work.